Hazel/Udp/UnityUdpClientConnection.cs

```diff
@@ -225,12 +236,7 @@ protected override bool SendDisconnect(MessageWriter data = null)
 
             try
             {
-                socket.SendTo(
-                    bytes,
-                    0,
-                    bytes.Length,
-                    SocketFlags.None,
-                    RemoteEndPoint);
+                this.WriteBytesToConnection(bytes, bytes.Length);
```

Comment:
I believe this is no longer guaranteed to put the final disconnect bytes on the wire during a dispose. WriteBytesToConnection is not expected to block, so if a block would occur, the dispose will likely happen first and cancel any unsent bytes. Using the sync method ensures that the bytes are sent prior to disposal and socket shutdown.

Reply:
Ah, this makes sense. The main problem here is ensuring the disconnect packet is properly encrypted and framed by a DTLS record.

I should be able to add a quick unit test for this case and work out something that prevents the issue you outlined.

Reply:
This should be addressed by https://github.com/willardf/Hazel-Networking/pull/11